### DISCUSSION TITLE
How to abort the current cypress test execution during run time

### DISCUSSION THREAD

**Discussion by SravanthiS123** [cypress-io/cypress#30634] [Questions and Help] [answered]
Hi,

Can anyone tell me how to abort the current cypress test (it block) execution after a certain few test steps by logging the results. I am aware of the command **cy.state('test').skip()** but it skips the current test run without logging any results of the test steps that were run so far. 

Sample code:
it("Sample test case", () => {
  for (let link of allLinks) {
    if (link.text() === 'Dashboard') {
      cy.state('test').skip();
    }
  }
})

**Comment by jennifer-shehane** [comment, marked as answer]
Cypress has no API to abort one test partway and keep it passed with the prior steps preserved, Mocha only allows passed, failed, or skipped/pending.

The approach that makes most sense is to not queue the remaining steps rather than abort a running one. Decide inside .then():

```js
it('Sample test case', () => {
  cy.get('a').then(($links) => {
    cy.log('completed initial steps')

    if ([...$links].some((el) => el.innerText.trim() === 'Dashboard')) {
      return // nothing else queued; test ends as passed, steps stay logged
    }

    cy.log('continuing remaining steps')
  })
})
```

If you’d rather the early exit be a real abort (failure), just throw new Error('...')
